concent_api/common/decorators.py

```diff
@@ -47,3 +51,15 @@ def wrapper(*args: Any, **kwargs: Any) -> None:
                 logging_level=LoggingLevel.ERROR)
             raise
     return wrapper
+
+
+def non_nesting_atomic(using: str=None, savepoint: bool=True) -> Atomic:
+    if (
+        settings.DETECT_NESTED_TRANSACTIONS and
+        get_connection(using if using is not None else DEFAULT_DB_ALIAS).in_atomic_block
```

Comment:
@rwrzesien, what would be the case when there is no `using`? Should we even bother with the default value here?

Reply:
I thought about it as well. If there is no `using` `DEFAULT_DB_ALIAS` is taken and it is `str: 'default'`. We are not using `default` database, only `storage` and `control`. 

But in django, if there is no `using` it means that this concerns `default` database.  So i think we will not need this now, but it is logic and coherent with django.

Reply:
> And generally big +1 for this decorator, it took me a while to understand senior's code.

Go ahead and check `transaction.atomic`, I have the idea from there ;)

> what would be the case when there is no using? Should we even bother with the default value here?

Exactly like @pawelkisielewicz wrote, we shouldn't bother because we use it like `transaction.atomic`, in which we have always specified the database, but in case when we don't want to, we don't have to add anything extra to the decorator's code itself.

Reply:
Quite frankly, I would still be for not having default value here, i.e. require that DB name is explicitly provided to the function. I think we don't have to exactly copy Django's logic here, as it is more confusing than just simply requiring parameter.

Reply:
What is better - having more options to use this decorator, or less options? Because in the solution you propose you limit options only to usages with explicitly provided database name.

Reply:
:+1: done

Reply:
Looking at it again now - I see that my biggest issue and the reason I agreed with @dybi was the use of a default value. I think that default values are overused in general and it's often better to force the user of the function to specify a value explicitly than have a default that is not obvious or might change, silently breaking other code.

For me forcing the programmer to spell out the name of the database is a feature for us - because we have two databases and we have to always be very careful about which one we use.

Reply:
> For me forcing the programmer to spell out the name of the database is a feature for us - because we have two databases and we have to always be very careful about which one we use.

That makes sense, and it makes even more that neither of those two is a default one. But I always lean towards the approach that developers are conscious when using the code which have defaults like this, and know when to use it or not. But I will not argue if this approach is always correct.